contracts/periphery/contracts/s2s/examples/erc20-mapping/Backing.sol

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@darwinia/contracts-utils/contracts/Scale.types.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "../../xapps/PangolinXApp.sol";
+import "../../calls/PangolinCalls.sol";
+
+pragma experimental ABIEncoderV2;
+
+interface IIssuing {
+    function issueFromRemote(
+        address mappedToken,
+        address recipient,
+        uint256 amount
+    ) external;
+}
+
+contract Backing is PangolinXApp {
```

Comment:
Can we not explicitly specify which chain to use.

Reply:
And I think it's better to use a combination than to use inheritance, especially since the SDK here can be provided entirely using Lib.

Reply:
Let me think about it.

Reply:
I think it is easy to do this. 
Use the code from SmartChainXApp(sendMessage, etc) directly.  It is a way for more familiar developers.